### PR TITLE
Homework1 Task5 find_maximal_subarray_sum

### PR DESCRIPTION
**Pull request by IuliiaLanchuk**

### CODE REVIEW

**Comment by Cjkjvfnby** [comment]
Please take into account:

- If I'm not assigned as a reviewer, I don't have any notification. 
- When I request changes, I automatically unsubscribed from the review and you need to request a review once again.

**Comment by IuliiaLanchuk** [comment]
Sorry, i really forgot to add you as a reviewer. Thank you for your notice.

File: homework1/task05/max_subarr_sum/task05.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    result = 16
+"""
+
+
+from typing import List
+
+
+def find_maximal_subarray_sum(nums: List[int], k: int) -> int:
+    """
+    I'm find max subarray sum, where k is the length of subarray.
```
Comment (Cjkjvfnby) [inline homework1/task05/max_subarr_sum/task05.py]: Please deal with the docstring.

File: homework1/task05/max_subarr_sum/task05.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    :param nums: input list of integers, k: input int
+    :return: int
+    """
+    max_sum = 0
+    for i in range(len(nums)):
+        for j in range(1, k + 1):
+            sub_list = nums[i : i + j]
```
Comment (Cjkjvfnby) [inline homework1/task05/max_subarr_sum/task05.py]: Please change this line to  `sub_list = nums[i:j]` and adjust other code to make it work.

File: homework1/task05/tests/test_max_subarr_sum.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                3,
+                -1,
+                -3,
+                -4,
+                23,
+                0,
+                -15,
+            ],
+            7,
+            51,
```
Comment (Cjkjvfnby) [inline homework1/task05/tests/test_max_subarr_sum.py]: The test should prove that your code is correct.  To prove it you need to take requirements and input and calculate the result (without using your code.) 

If you execute code and get results, you don't test that code do something, you test what value produced by code equals the value produced by code.  

Please use the simple case.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: homework1/task05/max_subarr_sum/task05.py (hunk 1 above)
Comment (IuliiaLanchuk) [reply homework1/task05/max_subarr_sum/task05.py]: Fixed

File: homework1/task05/max_subarr_sum/task05.py (hunk 2 above)
Comment (IuliiaLanchuk) [reply homework1/task05/max_subarr_sum/task05.py]: I removed one cycle _for_ and thus reduce code complexity

File: homework1/task05/tests/test_max_subarr_sum.py (hunk 3 above)
Comment (IuliiaLanchuk) [reply homework1/task05/tests/test_max_subarr_sum.py]: Fixed. Now I understand that simple tests are better than bigger ones. I will do simple tests next time. As all my pull requests  were done at the same time, I have same complex tests in every task :D.

File: homework1/task05/tests/test_max_subarr_sum.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+
+@pytest.mark.parametrize(
+    ("a", "k", "expected_result"),
+    [
+        ([1, 3, -1, -3, 5], 2, 4),
+        ([], 5, -sys.maxsize),
+        ([1, 3], 0, 0),
+        ([-1, -3, -1, -3, -5], 2, -4),
+    ],
```
Comment (Cjkjvfnby) [inline homework1/task05/tests/test_max_subarr_sum.py:13]: Let's test `[3, 3, -6, 1],  3,  6`

File: homework1/task05/max_subarr_sum/task05.py (hunk 2 above)
Comment (Cjkjvfnby) [reply homework1/task05/max_subarr_sum/task05.py]: > I removed one cycle for and thus reduce code complexity

Don't throw the baby out with the bathwater.

File: homework1/task05/tests/test_max_subarr_sum.py (hunk 4 above)
Comment (IuliiaLanchuk) [reply homework1/task05/tests/test_max_subarr_sum.py:13]: 6 - is incorrect value. 0 is correct.

Comment (IuliiaLanchuk) [reply homework1/task05/tests/test_max_subarr_sum.py:13]: Sorry, you are right! 6 is correct! My code works incorrect. Thanks. I will change code.

File: homework1/task05/max_subarr_sum/task05.py (hunk 2 above)
Comment (IuliiaLanchuk) [reply homework1/task05/max_subarr_sum/task05.py]: Done

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**